lib/_http_outgoing.js

```diff
@@ -426,6 +430,39 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 };
 
 
+// Returns an array of the names of the current outgoing headers.
+OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
+  return Object.keys(this._headers);
+};
+
+
+// Returns somewhat shallow copy of current outgoing headers. Array header
+// values will be cloned but no deeper cloning will occur.
+OutgoingMessage.prototype.getHeaders = function getHeaders() {
+  const headers = this._headers;
+  const keys = Object.keys(headers);
+  const ret = new OutgoingHeaders();
+  for (var i = 0; i < keys.length; ++i) {
+    const key = keys[i];
+    const val = headers[key][1];
+    if (val instanceof Array)
+      ret[key] = val.slice(0);
+    else
+      ret[key] = val;
+  }
+  return ret;
+};
+
+
+OutgoingMessage.prototype.hasHeader = function hasHeader(name) {
+  if (typeof name !== 'string') {
+    throw new TypeError('"name" argument must be a string');
+  }
+
+  return (this._headers && !!this._headers[name.toLowerCase()]);
```

Comment:
Should the `!!` be moved outside of the parens to guarantee a boolean is always returned?